api/main_endpoints/routes/UserSearch.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const User = require('../models/User.js');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid,
+} = require('../util/token-functions');
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+} = require('../../util/constants').STATUS_CODES;
+const membershipState = require('../../util/constants').MEMBERSHIP_STATE;
+
+const ROWS_PER_PAGE = 20;
+
+// Search for all members using either first name, last name or email
+router.post('/shortcutsearchusers', async function(req, res) {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req, membershipState.OFFICER)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const limit = Math.min(Number(req.body.limit) || ROWS_PER_PAGE, ROWS_PER_PAGE);
+  let maybeOr = {};
+  const query = req.body.query;
+  if (query) {
+    const parts = query.trim().split(/\s+/);
+    const first = parts[0];
+    const last = parts[parts.length - 1];
+    const nameRegex = new RegExp(`^${first}.*${last}`, 'i');
+
+    maybeOr = {
+      $or: [
+        {
+          $expr: {
+            $regexMatch: {
+              input: { $concat: ['$firstName', ' ', '$lastName'] },
```

Comment:
see other comment about regex change, we can just do
```suggestion
              input: { $concat: ['$firstName', '$lastName'] },
```

Reply:
Done!